File: fix-vendor-dashboard.js
const fs = require('fs');

const filePath = 'client/src/pages/VendorDashboard.tsx';

// Read the file
let content = fs.readFileSync(filePath, 'utf8');

// Add the CouponsTab import after the toast import
if (!content.includes('import { CouponsTab }')) {
    content = content.replace(
        'import { toast } from "sonner";',
        'import { toast } from "sonner";\nimport { CouponsTab } from "@/components/dashboard/CouponsTab";'
    );
}

// Update the activeTab type to include "coupons"
content = content.replace(
    'return (params.get("tab") as "overview" | "products" | "orders" | "settings" | "collections" | "categories") || "overview";',
    'return (params.get("tab") as "overview" | "products" | "orders" | "settings" | "collections" | "categories" | "coupons") || "overview";'
);

// Write the file back
fs.writeFileSync(filePath, content, 'utf8');

console.log('✅ VendorDashboard.tsx updated successfully!');
console.log('- Added CouponsTab import');
console.log('- Updated activeTab type to include "coupons"');

File: fix_vendor_dashboard.py
import sys

file_path = sys.argv[1]

with open(file_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Find and update the import section (after line 39: import { toast } from "sonner";)
for i, line in enumerate(lines):
    if 'import { toast } from "sonner";' in line:
        # Check if CouponsTab import already exists
        if i + 1 < len(lines) and 'CouponsTab' not in lines[i + 1]:
            lines.insert(i + 1, 'import { CouponsTab } from "@/components/dashboard/CouponsTab";\n')
        break

# Find and update the activeTab type definition
for i, line in enumerate(lines):
    if '"overview" | "products" | "orders" | "settings" | "collections" | "categories"' in line and '"coupons"' not in line:
        lines[i] = line.replace(
            '"overview" | "products" | "orders" | "settings" | "collections" | "categories"',
            '"overview" | "products" | "orders" | "settings" | "collections" | "categories" | "coupons"'
        )
        break

with open(file_path, 'w', encoding='utf-8') as f:
    f.writelines(lines)

print("File updated successfully")
